lib/src/http/auto_retry.dart

```diff
@@ -0,0 +1,105 @@
+library w_transport.src.http.auto_retry;
+
+import 'package:w_transport/src/http/base_request.dart';
+import 'package:w_transport/src/http/finalized_request.dart';
+import 'package:w_transport/src/http/request_exception.dart';
+import 'package:w_transport/src/http/requests.dart';
+import 'package:w_transport/src/http/response.dart';
+
+typedef /* Future<bool> | bool */ RetryTest(
+    FinalizedRequest request, BaseResponse response);
```

Comment:
no need for this distinction here.  just assume the return type is `Future<bool>`.  if you do that, the async keyword is all that's necessary for a simple function to match the signature:
```
Future<bool> tryIt() async {
  return false;
}
```